mpa/cls/stage.py

```diff
@@ -204,17 +204,20 @@ def configure_task(cfg, training, model_meta=None, **kwargs):
 
                 if not cfg.model.get('multilabel', False) and not cfg.model.get('hierarchical', False):
                     efficient_mode = cfg['task_adapt'].get('efficient_mode', True)
-                    gamma = 2 if efficient_mode else 3
                     sampler_type = 'balanced'
 
                     if len(set(model_classes) & set(dst_classes)) == 0 or set(model_classes) == set(dst_classes):
                         cfg.model.head.loss = dict(type='CrossEntropyLoss', loss_weight=1.0)
                     else:
+                        cls_num_list = [train_data_cfg.num_data[data_cls] for data_cls in dst_classes]
+                        if cfg.model.head.type == 'NonLinearClsHead':
+                            cfg.model.head.type = 'NonLinearIBLossHead'
+                        else:
+                            cfg.model.head.type = 'IBLossHead'
                         cfg.model.head.loss = ConfigDict(
-                            type='SoftmaxFocalLoss',
-                            loss_weight=1.0,
-                            gamma=gamma,
-                            reduction='none',
+                            type='IBLoss',
```

Comment:
Hardcoded loss is also confusing: why not implicitly add this to `.yaml` configs? Now one can use softmax only, but after these changes only `IBLoss` will be available. What if someone wants to experiment with softmax and add it to a config? The changes in the config won't be effective.

Reply:
I have no choice but hardcode this part to apply the IB loss only in incremental setting. The current code is already hardcoded for enabling focal loss for incremental setting(where new classes are added, thus it checks the difference between label schema of model and dataset). I agree that this hardcode part is somewhat problematic for experiments, so should be dealt when we are in refactoring stage next sprint.